_posts/2019-10-14-improving-prometheus-metrics.md

```diff
@@ -0,0 +1,134 @@
+But not as a JMX or Prometheus metric.
+
+Especially the consumer lag is very important metric because it allows you to monitor how big is the delay between the messages consumed by your consumers and the edge of your topic.
+When the lag is too big, it indicates that the consumers are slower than the producers and they are falling behind.
+If you applications are supposed to work in near time, you might need to do with it.
+
+Having the consumer lag in your Grafana dashboards and being able to configure alerts based on it will make it much easier to monitor your Kafka based applications.
+That is the reason why we added support for [Kafka Exporter](https://github.com/danielqsj/kafka_exporter).
+Kafka Exporter is a great open source project from [Daniel Qian](https://github.com/danielqsj) and other contributors - thanks for all your work.
+Strimzi just integrates it into our own Kafka deployments.
+Kafka Exporter connects to Kafka as a client and collects different information about topics, partitions and consumer groups.
+It then exposes this information as a Prometheus metric endpoint.
+
+Deploying Kafka Exporter with Strimzi is easy.
+All you need to do is to add it into a `Kafka` custom resource:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  # ...
+  kafkaExporter: {}
+```
+
+Strimzi will create a new deployment with the Kafka Exporter and configure it to connect to the Kafka cluster.
+You do not need to create any certificates or configure where it should connect.
+But if you need, you can configure the consumer groups or the topics which it should monitor:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  # ...
+  kafkaExporter:
+    topicRegex: "important-topic-.*"
+    groupRegex: "important-group-.*"
+```
+
+You can find all the available configuration option in our [API reference](https://strimzi.io/docs/latest/full.html#type-KafkaExporterSpec-reference).
+
+Kafka Exporter doesn't provide only the consumer lag metric.
+It also provides a lot of additional details about consumer groups and topics.
+* Information about message consumption rates.
+* Latest offsets per consumer group.
+* Latest and oldest offset for topics.
+* Information about partitions which do not have their leader on the preferred node.
+All these metrics can be used in alerts and dashboards.
+
+[![Kafka Exporter dashboard]({{ "/assets/2019-10-14-kafka-exporter-dashboard.png" }})](/assets/2019-10-14-kafka-exporter-dashboard.png)
+
+Once you have the Kafka Exporter deployed, you can start scraping the metrics it provides.
+We also provide a new Grafana dashboard and alert rules which work with the new metrics.
+In 0.14.0 the dashboard we have is fairly basic.
+In 0.15.0 we will release an improved Grafana dashboard which uses more of the different metrics provided by the Kafka Exporter.
```

Comment:
```suggestion
For 0.15.0 we are working on an improved Grafana dashboard which uses more of the different metrics provided by the Kafka Exporter.
```

Reply:
This is already done. I think I would prefer the keep my form to me this more clear so that the users can go and use the master version.